app/images.py

```diff
@@ -26,7 +31,83 @@ def run(self):
             time.sleep(self.interval)
 
             print('Image Pull Complete!')
-  
+
+def resize_image(filename):
+    img = Image.open(filename)
+
+    file_type = os.path.splitext(filename)[-1]
+    out_file = 'static/img/resized_image%s' % file_type
+
+    # Check to see which is bigger (width or height)
+    
+    if img.size[0] >= img.size[1] : # width > height
+        basewidth = config['screen_size']['width']
+        
+        wpercent = (basewidth / float(img.size[0]))
+        hsize = int((float(img.size[1]) * float(wpercent)))
+        img = img.resize((basewidth, hsize), PIL.Image.ANTIALIAS)
+        img.save(out_file)
+    else:
+        baseheight = config['screen_size']['height']
+        
+        hpercent = (baseheight / float(img.size[1]))
+        wsize = int((float(hpercent) * img.size[0]))
+        img = img.resize((wsize, baseheight), PIL.Image.ANTIALIAS)
+        img.save(out_file)
+    
+    return out_file
+
+def download_image(url):
+    response = requests.get(url, stream=True)
+    
+    # Get file extension of image
+    img_type = response.headers['Content-Type']
+    img_type = img_type.replace('image/', '')
+
+    filename = 'static/img/background.%s' % img_type 
+
+    with open(filename, 'wb') as out_file:
+        shutil.copyfileobj(response.raw, out_file)
+
+    return resize_image(filename)
+
+def generate_return_data(path, data):
+    # Generate 'path' without the root folder, and filename
+    s_path = path.split('/')
+    del s_path[0]
+    del s_path[0]
+    del s_path[-1]
+
+    path = ''
+    for s in s_path:
+        path += s
+        path += '/'
```

Comment:
I'm sure there is a utility that you could use to extract the path into it's elements rather than relying on the path separator.